ipa-core/src/protocol/basics/mod.rs

```diff
@@ -18,32 +18,44 @@ pub use share_known_value::ShareKnownValue;
 
 use crate::{
     const_assert_eq,
-    ff::{boolean::Boolean, PrimeField},
+    ff::{boolean::Boolean, ec_prime_field::Fp25519, PrimeField},
     protocol::{
-        context::{
-            Context, SemiHonestContext, UpgradedMaliciousContext, UpgradedSemiHonestContext,
-        },
+        context::{Context, SemiHonestContext, UpgradedSemiHonestContext},
         ipa_prf::{AGG_CHUNK, PRF_CHUNK},
+        prss::FromPrss,
     },
     secret_sharing::{
-        replicated::{
-            malicious::{AdditiveShare as MaliciousReplicated, ExtendableField},
-            semi_honest::AdditiveShare,
-        },
-        FieldSimd, SecretSharing, SharedValue, Vectorizable,
+        replicated::semi_honest::AdditiveShare, FieldSimd, SecretSharing, SharedValue, Vectorizable,
     },
     sharding::ShardBinding,
 };
 
+/// Basic suite of MPC protocols for vectorized data.
+///
+/// A previous version of `BasicProtocols` additionally required `Reshare` and `ShareKnownValue`,
+/// but those are omitted here because they are not vectorized. (`ShareKnownValue` has the
+/// difficulty of resolving `V` vs. `[V; 1]` issues for the known value type. `Reshare` hasn't been
+/// attempted.)
 pub trait BasicProtocols<C: Context, V: SharedValue + Vectorizable<N>, const N: usize = 1>:
-    SecretSharing<V>
-    + Reshare<C>
-    + Reveal<C, N, Output = <V as Vectorizable<N>>::Array>
-    + SecureMul<C>
-    + ShareKnownValue<C, V>
+    SecretSharing<V> + Reveal<C, N, Output = <V as Vectorizable<N>>::Array> + SecureMul<C> + FromPrss
+{
+}
+
+// For PRF test
+impl<'a, B: ShardBinding> BasicProtocols<UpgradedSemiHonestContext<'a, B, Fp25519>, Fp25519>
+    for AdditiveShare<Fp25519>
+{
+}
+
+impl<'a, B: ShardBinding>
+    BasicProtocols<UpgradedSemiHonestContext<'a, B, Fp25519>, Fp25519, PRF_CHUNK>
```

Comment:
Why this type of context?

Reply:
The idea here is that it should be supported for any context that satisfies `C: UpgradedContext`, although the combinations may not all be implemented if we don't need them anywhere.